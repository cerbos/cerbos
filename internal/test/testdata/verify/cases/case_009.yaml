description: Non-duplicate test (#1490)
