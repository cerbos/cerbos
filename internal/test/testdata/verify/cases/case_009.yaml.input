-- testdata/principals.yaml --
---
principals:
  john:
    policyVersion: '20210210'
    id: john
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    policyVersion: '20210210'
    kind: leave_request
    id: XX125
    attr: &attr
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design

-- testdata/auxdata.yaml --
---
auxData:
  myJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]

-- suite_test.yaml --
---
name: TestSuite
description: Tests
tests:
  - name: Test 1
    input:
      principals:
        - john
      resources:
        - john_leave_request
      actions:
        - view:public
      auxData: myJWT
    expected:
      - principal: john
        resource: john_leave_request
        actions:
          view:public: EFFECT_ALLOW

  - name: Test 2
    input:
      principals:
        - john
      resources:
        - john_leave_request
      actions:
        - view:public
      auxData: myJWT
    expected:
      - principal: john
        resource: john_leave_request
        actions:
          view:public: EFFECT_ALLOW
