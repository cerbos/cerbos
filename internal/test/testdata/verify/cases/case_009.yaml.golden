{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_PASSED",
        "testsCount": 2,
        "resultCounts": [
          {
            "result": "RESULT_PASSED",
            "count": 2
          }
        ]
      },
      "testCases": [
        {
          "name": "Test 1",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "john_leave_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        },
        {
          "name": "Test 2",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "john_leave_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_ALLOW"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ],
      "description": "Tests"
    }
  ],
  "summary": {
    "overallResult": "RESULT_PASSED",
    "testsCount": 2,
    "resultCounts": [
      {
        "result": "RESULT_PASSED",
        "count": 2
      }
    ]
  }
}
